### PR TITLE
Restauration

### LINKED ISSUE #80: Restauration - Menu
depends on https://github.com/epfl-idevelop/elements/issues/103

### PR DESCRIPTION
**Pull request by jminguely**
- Dropdown with multiple select (checkboxes)
- Table with tablesaw for sorting and responsiveness

Close #80

**Note about Tablesaw**
I propose to change the inclusion of Tablesaw with a SCSS compilation in the styleguide instead of a concatenation of the CSS file. Mainly to synchronize the tablesaw breakpoint (used to switch the table) with the same Bootstrap is using for his `md` view. We now need to check all usage of tablesaw and nest all `table.tablesaw` inside a `div.tablesaw-container`

### CODE REVIEW

**Comment by tonifisler** [comment]
This select plugin doesn't look like the most accessible one... :/

**Comment by knuch** [comment]
I slightly changed markup for tags to be more "affordant" like in the design and ux.

concerns I have:
- maybe the tech behind this restauration tool won't make it possible to use the selected js plugin which seems to be expecting live reload of the bottom list. If that's the case, we will have to add a validation button inside the dropdown like in the design, and maybe change plugin. 
- a11y-wise, I think we would need some advice on how to make the plugin accessible. Once more, it depends on how much we can ask for a markup modification.

OK to be merged in the actual state for me, but we need more infos ping @zufrieden

**Comment by tonifisler** [comment]
I updated the table styles to avoid this issue:

table_spacing

File: assets/components/organisms/restauration/restauration.twig [hunk 1]
```diff
@@ -66,19 +66,19 @@
...
                 <div class="col-6 text-right">11.90</div>
               </div>
               <div class="row">
-                <div class="col-6 text-primary">V</div>
+                <div class="col-6 text-primary"><abbr title="Vieux">V</abbr></div>
                 <div class="col-6 text-right">14.00</div>
               </div>
               <div class="row">
-                <div class="col-6 text-primary">D</div>
+                <div class="col-6 text-primary"><abbr title="Daltonien">D</abbr></div>
```
Comment (tonifisler) [inline assets/components/organisms/restauration/restauration.twig]: Needs real (or plausible) wording before we merge this.

Comment (jminguely) [reply assets/components/organisms/restauration/restauration.twig]: I found the legend at the bottom of the page 😉
D -> Doctorants

**Comment by Yago** [comment]
@tonifisler When you approve something, feel free to merge or at least, request somebody esle's review 😉

**Comment by tonifisler** [comment]
🤔 Your review is already requested, but yeah I would love some review since I also worked on this for some bits!

**Comment by Yago** [comment]
And @knuch too, like almost all our PRs on this project, but ok, I will review it too 😉

File: assets/components/atoms/select/select-multiple.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* globals $ */
+
+export default () => {
+  $('.select-multiple').each((index, element) => {
+    $(element).multipleSelect({
+      placeholder: $(element).attr('data-placeholder') ? $(element).attr('data-placeholder') : '',
```
Comment (Yago) [inline assets/components/atoms/select/select-multiple.js]: Holly molly 😅 `$(element).attr('data-placeholder') || ''` is enough, no need of repetition

File: assets/components/organisms/restauration/restauration.twig [hunk 3]
```diff
@@ -1 +1,104 @@
...
+    </li>
+  {% endfor %}
+</ul>
+
+<div class="tab-content py-3">
+  {% for day in restauration %}
+  <div class="tab-pane fade show {% if loop.first %}active{% endif %}" id="{{ day|lower }}" role="tabpanel" aria-labelledby="{{ day|lower }}-tab">
+    <div class="row my-3">
+      <div class="col-md-4">
+        <select class="select-multiple mb-3" multiple="multiple" data-placeholder="Restaurant">
```
Comment (Yago) [inline assets/components/organisms/restauration/restauration.twig]: Simply include your `select/select-multiple` component insteand of duplicating code 😉

File: assets/components/atoms/select/select.scss [hunk 4]
```diff
@@ -1 +1,139 @@
...
+      background-color: $dropdown-link-hover-bg;
+      color: $dropdown-link-hover-color;
+    }
+
+    &.optgroup {
+      font-weight: $font-weight-bold;
+    }
+  }
+
+  input[type="checkbox"] {
```
Comment (Yago) [inline assets/components/atoms/select/select.scss]: Not possible to extend B4 custom checkbox for better consistency ?

File: assets/components/organisms/table/table.scss [hunk 5]
```diff
@@ -0,0 +1,189 @@
+@charset 'utf-8';
+
+.table {
+  thead {
+    th {
```
Comment (Yago) [inline assets/components/organisms/table/table.scss]: unnecessary nesting

File: assets/components/organisms/table/table.scss [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+      .tablesaw-cell-label {
+        display: inline-block;
+        min-width: 33%;
+        color: $body-color;
+      }
+    }
+  }
+
+  tfoot {
+    td {
```
Comment (Yago) [inline assets/components/organisms/table/table.scss]: unnecessary nesting

File: assets/components/atoms/select/select.scss (hunk 4 above)
Comment (tonifisler) [reply assets/components/atoms/select/select.scss]: I would have loved that too, but the plugin doesn't allow to change its markup output. :/

File: assets/components/organisms/table/table.scss (hunk 6 above)
Comment (tonifisler) [reply assets/components/organisms/table/table.scss]: What do you mean? A class should be created instead?

**Comment by Yago** [comment]
⚠️ Doesn't match the components tree anymore. Fix before merging !

**Comment by jminguely** [comment]
Thanks for you review and help on this PR. 

I've proceeded with the requested correction and merge the current dev branch. That way the components tree should but up to date with the project.

**Pull request merged**